UnattendResources/Logon.ps1

```diff
@@ -241,6 +241,18 @@ function Run-CustomScript {
     }
 }
 
+function Install-VMwareTools {
+    $Host.UI.RawUI.WindowTitle = "Installing VMware tools..."
+    $vmwareToolsInstallArgs = "/s /v /qn REBOOT=R /l $ENV:Temp\vmware_tools_install.log"
+    if (Test-Path $resourcesDir) {
+        $vmwareToolsPath = Join-Path $resourcesDir "\VMware-tools.exe"
+    }
+    Start-Process -FilePath $vmwareToolsPath -ArgumentList $vmwareToolsInstallArgs -Wait -verb runAS

```

Comment:
here the last exit code won't work
You need to check it like this:
$p  = Start-Process -FilePath $vmwareToolsPath -ArgumentList $vmwareToolsInstallArgs `
                               -PassThru -Wait -verb runAS
if ($p.ExitCode) {
    throw "VMware tools setup failed!" 
}